### PR TITLE
feature(unlock-js): support for actual purchase thru CardPurchaser contract

### PR DESCRIPTION
**Pull request by julien51**
# Description

This adds actual support for purchases through the card purchaser!

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: packages/unlock-js/src/walletService.ts [hunk 1]
```diff
@@ -1022,18 +1028,17 @@ export default class WalletService extends UnlockService {
     params: { network: number }
   }) {
     const networkConfig = this.networks[network]
-    const UnlockSwapPurchaser = networkConfig?.swapPurchaser
-    if (!UnlockSwapPurchaser) {
+    const swapPurchaserAddress = networkConfig?.swapPurchaser
```
Comment (julien51) [inline packages/unlock-js/src/walletService.ts:1031]: renamed for consistency

File: packages/unlock-js/src/walletService.ts [hunk 2]
```diff
@@ -1074,7 +1079,7 @@ export default class WalletService extends UnlockService {
       from: await this.signer.getAddress(),
       to: ethers.utils.getAddress(cardPurchaserAddress),
       value,
-      validAfter: now,
+      validAfter: 0,
```
Comment (julien51) [inline packages/unlock-js/src/walletService.ts:1082]: This was creating issues in signature. There is actually no need to limit it.

File: packages/unlock-js/src/walletService.ts [hunk 3]
```diff
@@ -1102,4 +1107,31 @@ export default class WalletService extends UnlockService {
...
+  /**
+   * Performs a purchase using the CardPurchaser contract
+   * @param param0
+   * @returns
+   */
+  async purchaseWithCardPurchaser({
+    transfer,
+    purchase,
+    callData,
+  }: PurchaseWithCardPurchaserParams) {
```
Comment (searchableguy) [inline packages/unlock-js/src/walletService.ts:1120]: is this consistent with the argument format we use in above methods?

**Review by searchableguy — APPROVED** [review]

File: packages/unlock-js/src/walletService.ts (hunk 3 above)
Comment (julien51) [reply packages/unlock-js/src/walletService.ts:1120]: yes

**Pull request merged**